### PR TITLE
Core/Social: cleaunup, fix crash when friend or ignore get 0 guid and also double friend list

### PR DESCRIPTION
**Pull request by Viste**
## CHANGES PROPOSED:
-  cleanup
-  fix crash when friend or ignore get 0 guid( https://paste2.org/3dUy2Ud8 )
-  fix floating problem with double friend list(thx to TC team)

## ISSUES ADDRESSED:
don't find

## TESTS PERFORMED:
- build and game

## How to test the changes

-  Play with the Friend system (add/remove friends to your friend list), try login and logout between these actions too, make sure everything is fine

## Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Viste** [comment]
fuck you codacy :/

**Comment by Si1ker** [comment]
😿

File: src/server/game/Entities/Player/SocialMgr.cpp [hunk 1]
```diff
@@ -309,15 +307,15 @@ void SocialMgr::BroadcastToFriendListers(Player* player, WorldPacket* packet)
...
+PlayerSocial* SocialMgr::LoadFromDB(PreparedQueryResult result, uint64 guid)
 {
     PlayerSocial* social = &m_socialMap[guid];
     social->SetPlayerGUID(guid);
 
     if (!result)
         return social;
 
-    uint32 friendGuid = 0;
+    uint64 friendGuid = 0;
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/SocialMgr.cpp:318]: can't we declare `friendGuid` inside the `do` loop?

```c++
auto friendGuid = fields[0].GetUInt32();
```

Comment (Viste) [reply src/server/game/Entities/Player/SocialMgr.cpp:318]: auto is unsafe shit but okay

**Comment by FrancescoBorzi** [comment]
@azerothcore/testers this is quite important as it prevents crash, please test it

**Comment by JamesGent** [comment]
Compiled with docker and tried the following;

Add friends from same/other accounts.
logut.
login.
Remove friends.
logout
login
add friends again
logout
delete the characters
login

Not sure if this counts as a review as I have no idea what I was doing lmao

**Comment by Helias** [comment]
It is always good testing everything is related to the feature, thanks a lot! :wink:

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**